### PR TITLE
Type Stable Functional

### PR DESCRIPTION
**Pull request by eliascarv**
A small change to make the Functional type stable

### CODE REVIEW

**Comment by juliohm** [comment]
Simple and important improvement. Awesome.

As discussed over Zulip, I like the suggestion of allowing functor objects. Feel free to update the PR accordingly and add a test for this new feature.

**Comment by eliascarv** [comment]
This PR is ready for review @juliohm.

File: src/transforms/functional.jl [hunk 1]
```diff
@@ -7,12 +7,11 @@
...
 """
-struct Functional <: Colwise
-  func::Function
+struct Functional{F} <: Colwise
+  func::F
 end
 
-isrevertible(transform::Functional) =
-  !isnothing(inverse(transform.func))
+isrevertible(transform::Functional) = !isnothing(inverse(transform.func))
```
Comment (juliohm) [inline src/transforms/functional.jl:14]:
```suggestion
isrevertible(transform::Functional) =
  !isnothing(inverse(transform.func))
```

File: test/runtests.jl [hunk 2]
```diff
@@ -26,6 +26,18 @@ testfiles = [
...
+  coeffs::Vector{T}
+end
+Polynomial(args::T...) where {T<:Real} = Polynomial(collect(args))
+function (p::Polynomial)(x)
+  n = length(p.coeffs) - 1
+  sum(zip(p.coeffs, 0:n)) do (a, i)
+    a * x^i
+  end
+end
+
```
Comment (juliohm) [inline test/runtests.jl:40]: Please move to before the list of test files 👍🏽

File: test/transforms.jl [hunk 3]
```diff
@@ -358,6 +358,19 @@
     n, c = apply(T, t)
     @test t == n
     @test isrevertible(T) == false
+
+    # Functor tests
```
Comment (juliohm) [inline test/transforms.jl:362]:
```suggestion
    # functor tests
```

**Review by juliohm — COMMENTED** [review]
Awesome work. Just a small minor suggestion.

File: test/runtests.jl (hunk 2 above)
Comment (eliascarv) [reply test/runtests.jl:40]: OK

Comment (eliascarv) [reply test/runtests.jl]: Done

File: test/runtests.jl [hunk 4]
```diff
@@ -20,6 +20,18 @@ islinux = Sys.islinux()
...
+# for functor tests in Functional testset
+struct Polynomial{T<:Real}
+  coeffs::Vector{T}
+end
+Polynomial(args::T...) where {T<:Real} = Polynomial(collect(args))
+function (p::Polynomial)(x)
+  n = length(p.coeffs) - 1
+  sum(zip(p.coeffs, 0:n)) do (a, i)
+    a * x^i
+  end
```
Comment (juliohm) [inline test/runtests.jl:32]: What about

```julia
(p::Polynomial)(x) = sum(a*x^(i-1) for (i, a) in enumerate(p.coeffs))
```

Comment (eliascarv) [reply test/runtests.jl]: Done!

**Comment by juliohm** [comment]
Merging it 👍🏽

**Pull request merged**